llvm/lib/CodeGen/MachineVerifier.cpp

```diff
@@ -3697,6 +3697,9 @@ void MachineVerifier::verifyStackFrame() {
       if (I.getOpcode() == FrameSetupOpcode) {
         if (BBState.ExitIsSetup)
           report("FrameSetup is after another FrameSetup", &I);
+        if (!MRI->isSSA() && !MF->getFrameInfo().adjustsStack())
```

Comment:
We probably should have something better than isSSA. I've thought before we needed a FrameFinalized MachineFunction Property